src/main/clojure/clara/rules/platform.cljc

```diff
@@ -75,4 +75,10 @@
            ~@body
            (finally
              ~@(for [[tl] binding-pairs]
-                 `(.remove ~tl)))))))
\ No newline at end of file
+                 `(.remove ~tl)))))))
+
+#?(:clj
```

Comment:
I guess the best explanation is just https://clojurescript.org/about/differences#_macros but that doesn't really get directly to the point.

In cljs macros are done in a separate compilation stage and do need to be in a clj or cljc file. However, they are not a feature exclusive to a single platform - clj vs cljs.

The read conditionals are for forms that are exclusive to the particular platform, like `java.lang.String` vs `js/String`. 

`defmacro` is defined in both the clj and cljs compiler. Cljs macros must (of course) emit code that is meant for cljs compilation though.

Can't put macros in .cljs file because the cljs compiler expects the macros to have already be defined in a prior "compilation stage".